src/stream/src/executor/merge.rs

```diff
@@ -227,16 +305,41 @@ impl SelectReceivers {
         assert!(other.blocks.is_empty() && other.barrier.is_none());
         assert_eq!(self.actor_id, other.actor_id);
 
-        self.upstreams.extend(other.upstreams);
+        let mut new_identity_hint = self
+            .block_upstream_identities
+            .last()
+            .map(|&x| x + 1)
+            .unwrap_or_default();
```

Comment:
How about using the `Input::actor_id` as the index? So there's no need to maintain an index ourselves. Then it seems we can totally remove this field?

Reply:
> How about using the `Input::actor_id` as the index? So there's no need to maintain an index ourselves. Then it seems we can totally remove this field?

I have considered about this but I'm afraid we may have two same `actor_id` in the future

Reply:
This won't happen. The `remove` logic is relying on this as well.

Reply:
> How about using the `Input::actor_id` as the index? So there's no need to maintain an index ourselves. Then it seems we can totally remove this field?

resolved